rpc/transaction.go

```diff
@@ -66,21 +66,23 @@ func (t *TransactionType) UnmarshalJSON(data []byte) error {
 }
 
 // https://github.com/starkware-libs/starknet-specs/blob/a789ccc3432c57777beceaa53a34a7ae2f25fda0/api/starknet_api_openrpc.json#L1252
+//
+//nolint:lll
 type Transaction struct {
 	Hash                *felt.Felt      `json:"transaction_hash,omitempty"`
-	Type                TransactionType `json:"type"`
-	Version             *felt.Felt      `json:"version,omitempty"`
-	Nonce               *felt.Felt      `json:"nonce,omitempty"`
-	MaxFee              *felt.Felt      `json:"max_fee,omitempty"`
+	Type                TransactionType `json:"type" validate:"required"`
+	Version             *felt.Felt      `json:"version,omitempty" validate:"required"`
```

Comment:
Does it make sense to have `omitempty` tag for json while we also have `validate: "required"`? Can we not just have the json tag here?

Reply:
Validate tags are for incoming Transactions, json tags are used for marshaling only (ie for marshaling responses to the RPC requests we get). So they are not related.